theseus/embodied/robot/forward_kinematics.py

```diff
@@ -0,0 +1,146 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+#
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+import torch
+
+from typing import List, Optional
+
+from theseus.labs.lie_functional import se3
+from .robot import Robot
+from .link import Link
+from .joint import Joint
+
+
+# TODO: Add support for joints with DOF>1
+
+
+def ForwardKinematicsFactory(robot: Robot, link_names: Optional[List[str]] = None):
+    links: List[Link] = []
+
+    if link_names is None:
+        links = robot.links
+    else:
+        links = [robot.link_map[name] for name in link_names]
+
+    link_ids: List[int] = [link.id for link in links]
+
+    ancestors: List[Link] = []
+    joint_ids: List[int] = []
+    for link in links:
+        ancestors += [anc for anc in link.ancestors]
+        joint_ids += link.angle_ids
+    pose_ids = sorted(list(set([anc.id for anc in ancestors] + link_ids)))
+    joint_ids = sorted(list(set(joint_ids)))
+
+    def _forward_kinematics_helper(angles: torch.Tensor):
+        if angles.ndim != 2 or angles.shape[1] != robot.dof:
+            raise ValueError(
+                f"Joint angles for {robot.name} should be {robot.dof}-D vectors"
+            )
+
+        poses: List[Optional[torch.Tensor]] = [None] * robot.num_links
+        poses[0] = angles.new_zeros(angles.shape[0], 3, 4)
```

Comment:
Replace with `se3.new_identity(angles)`.

Reply:
Maybe we need more discussions about this the behavior of `se3.new_identity()`. For example, how to specify the dimensions.

Reply:
Is there any reasonable option other than the code below? 

```python
g = se3.new_identity(batch_size)
# g.shape = (batch_size, 3, 4)
```

Reply:
The `liegroup` has no `new_identity()` method. This is definitely in our TODO list and I will go back to this once `new_identity()` is added to Lie group.